src/axom/spin/examples/spin_bvh_two_pass.cpp

```diff
@@ -0,0 +1,309 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+// Axom includes
+#include "axom/core.hpp"
+#include "axom/mint.hpp"
+#include "axom/primal.hpp"
+#include "axom/spin.hpp"
+#include "axom/slic.hpp"
+#include "axom/quest.hpp"
+
+#include "axom/CLI11.hpp"
+#include "axom/fmt.hpp"
+
+namespace mint = axom::mint;
+namespace primal = axom::primal;
+namespace spin = axom::spin;
+namespace slic = axom::slic;
+namespace quest = axom::quest;
+
+using IndexType = axom::IndexType;
+using UMesh = mint::UnstructuredMesh<mint::SINGLE_SHAPE>;
+
+enum class ExecPolicy
+{
+  CPU,
+  OpenMP,
+  CUDA
+};
+
+const std::map<std::string, ExecPolicy> validExecPolicies {
+  {"seq", ExecPolicy::CPU},
+#ifdef AXOM_USE_OPENMP
+  {"omp", ExecPolicy::OpenMP},
+#endif
+#ifdef AXOM_USE_CUDA
+  {"cuda", ExecPolicy::CUDA}
+#endif
+};
+
+//------------------------------------------------------------------------------
+void initialize_logger()
+{
+  // initialize logger
+  slic::initialize();
+  slic::setLoggingMsgLevel(slic::message::Info);
+
+  // setup the logstreams
+  std::string fmt = "";
+  slic::LogStream* logStream = nullptr;
+
+  fmt = "[<LEVEL>]: <MESSAGE>\n";
+  logStream = new slic::GenericOutputStream(&std::cout, fmt);
+
+  // register stream objects with the logger
+  slic::addStreamToAllMsgLevels(logStream);
+}
+
+//------------------------------------------------------------------------------
+void finalize_logger()
+{
+  slic::flushStreams();
+  slic::finalize();
+}
+
+template <typename ExecSpace>
+void find_collisions_broadphase(const mint::Mesh* mesh,
```

Comment:
Clarification: What does "broadphase" mean in this context?

Since it's the only function call, would it be clearer to just call it `find_collisions` and add a comment to indicate what the function does?

Reply:
In the context of collision detection, broad-phase refers to a fast but conservative check to prune away the bulk of non-intersecting elements; the converse of that would be narrow-phase, where a more accurate intersection test is used on the remaining candidate intersections.

The corollary to our surface mesh self-intersection test example is that the broad phase is represented by the BVH query, and the narrow phase would be the subsequent triangle-triangle intersection tests. I could also add that from the mesh tester if it would make the example clearer.

Reply:
@kanye-quest actually, I think adding this explanation to a comment section at the top of the example code, would be great.

Reply:
> @kanye-quest actually, I think adding this explanation to a comment section at the top of the example code, would be great.

Thanks @rhornung67 -- yes, that's what I was hoping for.

Reply:
> The corollary to our surface mesh self-intersection test example is that the broad phase is represented by the BVH query, and the narrow phase would be the subsequent triangle-triangle intersection tests. I could also add that from the mesh tester if it would make the example clearer.

It might be helpful to have the narrow-phase part to show users how the broad-phase part can be used.
I.e. the goal of this example is to demonstrate how the 2-pass algorithm works, and the code currently only shows one of those passes.

When I saw the name `find_collisions_broadphase`, I was anticipating a second function that uses those results (e.g. `find_collisions_narrowphase`), or at least an explanation of how those results are expected to be used.